cluster/engine.go

```diff
@@ -53,6 +55,14 @@ type Engine struct {
 	eventHandler    EventHandler
 	healthy         bool
 	overcommitRatio int64
+	ScheduleQueue   *list.List
+}
+
+//ScheduledItem FIX ME : works for affinities as it needs the conatainer name and Image
+//Need to add more types to support all filters
+type ScheduledItem struct {
```

Comment:
Ideally (as in, in a perfect world) scheduled containers (not yet created) should be in `Node` and not `Engine`.

Here's the reasoning:
`Engine` is a type representing a Docker Engine, verbatim
`Node` is an internal structure of the scheduler on which it operates on. Right now it's pretty much just an Engine, but that's where we can put data types to be used by the scheduler and the scheduler only (how many failed attempts, bla bla).

Not sure if it's easy to do though.